### PR TITLE
re-enable matrices with NAs tests in X and layers

### PR DESCRIPTION
**Pull request by rcannood**

### CODE REVIEW

File: tests/testthat/test-roundtrip-X.R [hunk 1]
```diff
@@ -42,6 +42,12 @@ for (name in layer_names) {
     rownames(X) <- data$obs_names
     colnames(X) <- data$var_names
 
+    if (is.matrix(X) && any(is.na(X))) {
+      # is.na(value) <- NaN gets ignored
+      na_indices <- is.na(X)
+      X[na_indices] <- NaN
+    }
+
```
Comment (LouiseDck) [inline tests/testthat/test-roundtrip-X.R:50]: @rcannood I think this is something to take care if / when we want to add a reticulate backend, but not something we need to take into account now? The issue is only when writing using reticulate, when reading R does the 2^31 --> NA conversion automatically.

File: tests/testthat/test-roundtrip-X.R [hunk 2]
```diff
@@ -6,8 +6,8 @@ data <- generate_dataset_as_list(10L, 20L)
 layer_names <- names(data$layers)
 # TODO: re-enable these tests
 layer_names <- layer_names[!grepl("_dense", layer_names)]
-# TODO: re-enable these tests
-layer_names <- layer_names[!grepl("_with_nas", layer_names)]
+# # TODO: re-enable these tests
```
Comment (rcannood) [inline tests/testthat/test-roundtrip-X.R:9]: Remove if no longer needed ^^

```suggestion
```

File: tests/testthat/test-roundtrip-layers.R [hunk 3]
```diff
@@ -6,8 +6,8 @@ data <- generate_dataset_as_list(10L, 20L)
 layer_names <- names(data$layers)
 # TODO: Add denseMatrix support to anndata and anndataR
 layer_names <- layer_names[!grepl("_dense", layer_names)]
-# TODO: re-enable these tests
-layer_names <- layer_names[!grepl("_with_nas", layer_names)]
+# # TODO: re-enable these tests
+# layer_names <- layer_names[!grepl("_with_nas", layer_names)]
```
Comment (rcannood) [inline tests/testthat/test-roundtrip-layers.R:10]: Remove if no longer needed ^^

```suggestion
```

File: tests/testthat/test-roundtrip-layers.R [hunk 4]
```diff
@@ -69,8 +69,8 @@ for (name in layer_names) {
 r2py_names <- layer_names
 # TODO: rsparse gets converted to csparse by anndata
 r2py_names <- r2py_names[!grepl("rsparse", r2py_names)]
-# TODO: fix when this is working
-r2py_names <- r2py_names[!grepl("with_nas", r2py_names)]
+# # TODO: fix when this is working
+# r2py_names <- r2py_names[!grepl("with_nas", r2py_names)]
```
Comment (rcannood) [inline tests/testthat/test-roundtrip-layers.R:73]: Remove if no longer needed ^^

```suggestion
```

File: tests/testthat/test-roundtrip-X.R (hunk 1 above)
Comment (rcannood) [reply tests/testthat/test-roundtrip-X.R:49]: +1

**Comment by rcannood** [comment]
@lazappi Could you review this as well? :bow:

**Review by lazappi — APPROVED** [review]

**Pull request merged**